### ISSUE TITLE
Find in files using extenstion API

### ISSUE THREAD

**Issue by jaydenlin** [Microsoft/vscode#9049]
Hi VS code team,
I would like to make an extenstion using `Find in FIles` API.
Is there any way to trigger `Find in Files` via extension ?
I only find `workspace.findFiles`.
Thanks in advance.

**Comment by joaomoreno** [comment]
cc @jrieken

**Comment by jaydenlin** [comment]
Hi @jrieken,
Could you kindly provide any suggestion ?
Thank you.

**Comment by jrieken** [comment]
That hasn't been exposed yet - mainly because we might change the search engine we use behind the covers which might lead to API breakage...

**Comment by jaydenlin** [comment]
Hi @jrieken,
Okay.. I see.. Thank you for the reply.

**Comment by jrieken** [comment]
@roblourens Are we in a good enough state to expose our find in files functionalities for extension authors? Especially wrt regex-searching will we make more changes or is this future proof?

**Comment by roblourens** [comment]
Yeah, I think we could do this. Someone could implement something like https://github.com/Microsoft/vscode/issues/23931

**Comment by jrieken** [comment]
Yeah, like my alternative reference ui: https://github.com/Microsoft/vscode-extension-samples/blob/master/contentprovider-sample/README.md

**Comment by DHedgecock** [comment]
Could this be used to create an extension allowing for full page results? This editor is so awesome, but the current find results showing in the sidebar is difficult to use in my opinion. I'd like to make an extension for executing a search and opening the results in an editor (similar to Atom search flow).

Excited to see this land!

**Comment by IlyaBiryukov** [comment]
Ideally, the search API should match [SearchProvider](https://github.com/Microsoft/vscode/blob/b21e8d48fbe8367a45222f369154bb1da572d2d1/src/vs/vscode.proposed.d.ts#L325) API. Though, instead of registering a new search provider, the goal here is to be able to call it and get the search results. And that should work with the default provider that VSCode implements, when nobody else has actually registered one.

**Comment by IlyaBiryukov** [comment]
@jrieken when do you think it can land into VSCode?
